### PR TITLE
Performance examples improvements and minor fixes

### PR DESCRIPTION
**Pull request by marquezmanu**
- Addded throttle function to handleScrollEvent
- Catch property errors
- avoid calling getLoadingSpinnerHeight function unnecessary

### CODE REVIEW

File: examples/index.jsx [hunk 1]
```diff
@@ -11,42 +23,13 @@ var ListItem = React.createClass({
                 height: this.props.height,
                 lineHeight: this.props.lineHeight
             }
-        }>
-            List Item {this.props.index}
+        } key={this.props.key}>
```
Comment (marciocaraballo) [inline examples/index.jsx:26]: Where does this key come from ?

File: src/react-infinite.jsx [hunk 2]
```diff
@@ -287,9 +288,10 @@ var Infinite = React.createClass({
   },
 
   componentDidUpdate(prevProps: ReactInfiniteProps, prevState: ReactInfiniteState) {
-    this.loadingSpinnerHeight = this.utils.getLoadingSpinnerHeight();
-
     if (this.props.displayBottomUpwards) {
+      // this line was in the top of this function, moved to here for performance improvments
+      this.loadingSpinnerHeight = this.utils.getLoadingSpinnerHeight();
```
Comment (marciocaraballo) [inline src/react-infinite.jsx:293]: Can you confirm we don't need this when scrolling bottom ?

File: examples/index.jsx (hunk 1 above)
Comment (marquezmanu) [inline examples/index.jsx:26]: my bad, it should said this.props.index too

File: src/react-infinite.jsx (hunk 2 above)
Comment (marquezmanu) [inline src/react-infinite.jsx:293]: They only used this property in all the code when displayBottomUpwards === true, so i think that there were unnecessary logic triggered.

**Pull request merged**